src/engine/pieces/piece.ts

```diff
@@ -16,5 +29,65 @@ export default class Piece {
     public moveTo(board: Board, newSquare: Square) {
         const currentSquare = board.findPiece(this);
         board.movePiece(currentSquare, newSquare);
+        this.active = true;
+    }
+
+    public isValidMove(board: Board, nextSquare: Square) {
+
+        let pieceOnBoard = board.getPiece(nextSquare)
+        if (pieceOnBoard === undefined)
+            return SquareState.EMPTY_SQUARE;
+
+        if ((this.player != pieceOnBoard.player) && (pieceOnBoard.pieceType != PieceType.KING))
+            return SquareState.CAPTURABLE_PIECE_PRESENT;
+
+        return SquareState.UNCAPTURABLE_PIECE_PRESENT;
+    }
+
+    private getDirectionalMoves(board: Board, directionChanges: Array<PositionChange>)
+    {
+        let validMoves = new Array(0);
+        let currentSquare = board.findPiece(this);
+
+        directionChanges.forEach(directionChange => {
+            let nextSquare = currentSquare.moveBy(directionChange.row, directionChange.col);
+            let keepTesting = true;
+            while (board.isOnBoard(nextSquare) && keepTesting) {
+                switch (this.isValidMove(board, nextSquare)) {
+                    case SquareState.EMPTY_SQUARE:
+                        validMoves.push(nextSquare);
+                        break;
+                    case SquareState.CAPTURABLE_PIECE_PRESENT:
+                        validMoves.push(nextSquare);
+                        keepTesting = false;
+                        break;
+                    case SquareState.UNCAPTURABLE_PIECE_PRESENT:
+                        keepTesting = false;
+                        break;
+                }
+                nextSquare = nextSquare.moveBy(directionChange.row, directionChange.col);
+            }
+        })
+        return validMoves;
+    }
+
+    public getDiagonalMoves(board: Board): Array<Square>{
+        const directionChanges = [
+            {row: 1, col: 1}, 
+            {row: 1, col: -1}, 
+            {row: -1, col: 1}, 
+            {row: -1, col: -1}
+        ];
+        return this.getDirectionalMoves(board, directionChanges);
+    }
+
+    public getLateralMoves(board: Board): Array<Square>{
+        const directionChanges = [
+            {row: 1, col: 0}, 
+            {row: -1, col: 0}, 
+            {row: 0, col: 1}, 
+            {row: 0, col: -1}
+        ];
+        return this.getDirectionalMoves(board, directionChanges);
     }
```

Comment:
could you restyle this `directions` too, also should have said before but like I mentioned in the import we want to add spacing around objects so each line would be

`{ rowChange: -1, colChange: -1 }`